src/Aspire.Dashboard/wwwroot/js/app.js

```diff
@@ -53,15 +53,33 @@ function isScrolledToBottom(container) {
 }
 
 window.copyTextToClipboard = function (id, text, precopy, postcopy) {
-    let tooltipDiv = document.querySelector('fluent-tooltip[anchor="' + id + '"]').children[0];
+    let button = document.querySelector(`#${id}`);
+
+    // If there is a pending timeout then clear it. Otherwise the pending timeout will prematurely reset values.
+    if (button.dataset.copyTimeout) {
+        clearTimeout(button.dataset.copyTimeout);
+        delete button.dataset.copyTimeout;
+    }
+
+    let copyIcon = button.querySelector('.copy-icon');
+    let checkmarkIcon = button.querySelector('.checkmark-icon');
+    let tooltipDiv = document.querySelector(`fluent-tooltip[anchor="${id}"]`).children[0];
     navigator.clipboard.writeText(text)
         .then(() => {
             tooltipDiv.innerText = postcopy;
+            copyIcon.style.display = 'none';
+            checkmarkIcon.style.display = 'inline';
         })
         .catch(() => {
             tooltipDiv.innerText = 'Could not access clipboard';
         });
-    setTimeout(function () { tooltipDiv.innerText = precopy }, 1500);
+
+    button.dataset.copyTimeout = setTimeout(function () {
```

Comment:
Could you explain why all this code is JS and not blazor?

Reply:
I went with the existing code (changing the tooltip title) that already existed in JS.

I thought about doing everything in Blazor but there is UI lag to consider. If the dashboard is in ACA and you have a slow connection, then there could be a 1s+ delay between clicking copy and the confirmation that content was copied. JS will always be fast.